homeassistant/components/melcloud/sensor.py

```diff
@@ -165,7 +165,8 @@ class AtwZoneSensor(MelDeviceSensor):
 
     def __init__(self, api: MelCloudDevice, zone: Zone, measurement, definition):
         """Initialize the sensor."""
-        super().__init__(api, measurement, definition)
+        full_measurement = f"{measurement}-zone-{zone.zone_index}"
```

Comment:
Can we avoid adding the zone_index if the index is 0? That way it will minimize the impact of the breaking change.

Reply:
I'd rather check whether the device has more than 1 zones and add the suffix only in those cases. That way the 1 zone variant would not be impacted and the previously broken 2 zone setups would have consistent IDs. What do you think?

Reply:
The goal is to reduce the breaking changes. So if Home Assistant used to always receive index 0 first, that data is correct and it just ignored other zones. So by keeping index 0 in the old format, less will break.

Reply:
I implemented the `zone_index == 1` exception.

The solution I proposed has similar goals and effects.

- If a device has a single zone, the `unique_id` would remain the same.
- If a device has 2 zones, the sensors were unusable and there is nothing to break with the different `unique_id` generation.